clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/TransferMetadata.scala

```diff
@@ -32,19 +32,24 @@ trait TransferMetadata[M <: TransferMetadata[M]] { self: M =>
 
   /**
     * Return a copy of this object in which all files in `pathsToMove`
-    * have been moved into a destination directory.
+    * have been moved to the given destination.
     *
-    * Fails if `destination` doesn't end with '/'.
+    * If `pathsToMove` contains > 1 element, `destination` must end with '/'.
     */
-  def moveAllInto(destination: String): M
-
-  /**
-    * Returns copy of this object in which a singular file-path has
-    * been sent to a new destination.
-    *
-    * Fails if `pathsToMove` contains more than one element.
-    */
-  def setSinglePath(destination: String): M
+  def moveInto(destination: String): M = {
+    val pathMapper = if (destination.endsWith("/")) {
+      moveIntoDirectory(_: String, destination)
+    } else {
+      if (pathsToMove.length > 1) {
+        sys.error(
+          "Non-directory destination given for metadata with > 1 path to move"
+        )
+      }
+      (_: String) =>
+        destination
+    }
+    mapMove(pathMapper)
```

Comment:
This is better, but it looks like every use of `pathMapper` is to call `map` on an object. It seems like you should be able to abstract that as well, so callers don't need to call `map` explicitly.